src/Microsoft.AspNetCore.Mvc.Abstractions/ModelBinding/ModelStateDictionary.cs

```diff
@@ -174,7 +174,77 @@ public bool IsValid
 
         /// <summary>
         /// Adds the specified <paramref name="exception"/> to the <see cref="ModelStateEntry.Errors"/> instance
-        /// that is associated with the specified <paramref name="key"/>.
+        /// that is associated with the specified <paramref name="key"/>. If the maximum number of allowed
+        /// errors has already been recorded, ensures that a <see cref="TooManyModelErrorsException"/> exception is
+        /// recorded instead.
+        /// </summary>
+        /// <remarks>
+        /// This overload allows adding the <paramref name="exception"/> to the current <see cref="ModelStateDictionary"/>
+        /// when <see cref="ModelMetadata"/> is not available and/or the <paramref name="exception"/> has custom
+        /// or relevant information regarding the validation state.
```

Comment:
What does "custom or relevant information" mean? I'm not sure but suspect you mean
``` c#
/// ... or the exact <paramref name="exception"/> must be maintained for later use (even if it is for example a `FormatException`).
```